src/clj/puppetlabs/services/jruby/jruby_puppet_service.clj

```diff
@@ -15,18 +16,25 @@
 
 (trapperkeeper/defservice jruby-puppet-pooled-service
                           jruby/JRubyPuppetService
-                          [[:ConfigService get-config]
-                           [:ShutdownService shutdown-on-error]
-                           [:PuppetProfilerService get-profiler]
-                           [:PoolManagerService create-pool]]
+                          {:required [[:ConfigService get-config]
+                                      [:ShutdownService shutdown-on-error]
+                                      [:PuppetProfilerService get-profiler]
+                                      [:PoolManagerService create-pool]]
+                           :optional [MetricsService]}
+
   (init
     [this context]
     (log/info (i18n/trs "Initializing the JRuby service"))
-    (let [jruby-config (core/initialize-and-create-jruby-config
-                        (get-config)
+    (let [config (get-config)
+          metrics-registry (when-let [metrics-service (tk-services/maybe-get-service this :MetricsService)]
+                             {:metric-registry (metrics/get-metrics-registry metrics-service :puppetserver)
+                              :server-id (get-in config [:metrics :server-id])})
```

Comment:
I guess so. I don't really feel like it's that big a deal to be getting it out of the config, but it might be cleaner to have a service function.